### PR TITLE
refine BaseCard event handlers and children render

### PR DESCRIPTION
**Pull request by sjd78**
Based on work in #691 and #700, refine the `BaseCard`:

  - better description of the component in the class comment

  - event handlers are defined functions bound to `this` in the
    constructor, matching the style in the rest of the app

  - children are rendered as a function, element or any other
    node type via `renderChildren`

  - update components in **GridComponent.js** so they don't blindly pass props to a `div` tag and cause React errors

### CODE REVIEW

File: src/components/VmDetails/BaseCard.js [hunk 1]
```diff
@@ -90,7 +113,7 @@ class BaseCard extends React.Component {
             <CardEditButton tooltip={editTooltip} editEnabled={editing} onClick={this.clickEdit} />
           )}
 
-          {children({ isEditing: editing })}
+          <this.renderChildren isEditing={editing} />
```
Comment (bond95) [inline src/components/VmDetails/BaseCard.js]: This is not right calling react component. Please, assign it to capitalize variable before use.
https://reactjs.org/docs/jsx-in-depth.html#user-defined-components-must-be-capitalized

File: src/components/VmDetails/BaseCard.js [hunk 2]
```diff
@@ -118,7 +141,7 @@ BaseCard.propTypes = {
   onStartEdit: PropTypes.func,
   onCancel: PropTypes.func,
   onSave: PropTypes.func,
-  children: PropTypes.func,
+  children: PropTypes.oneOfType([ PropTypes.func, PropTypes.ndoe ]),
```
Comment (bond95) [inline src/components/VmDetails/BaseCard.js]: Not `ndoe`, but `node`. Actually I think it can be set as `isRequired`.

Comment (sjd78) [reply src/components/VmDetails/BaseCard.js]: :man_facepalming:  I'll fix both.

File: src/components/VmDetails/BaseCard.js (hunk 1 above)
Comment (sjd78) [reply src/components/VmDetails/BaseCard.js]: Makes sense.  I'll make that change.

**Pull request merged**